src/muxer/mplex/mplex_stream.cpp

```diff
@@ -136,7 +136,8 @@ namespace libp2p::connection {
       return cb(Error::INVALID_ARGUMENT);
     }
     if (is_writing_) {
-      return cb(Error::IS_WRITING);
+      write_queue_.emplace_back(in, bytes, cb);
```

Comment:
It looks like queue access shold be secured with mutex.
The same is applicable to yamux too.

Reply:
Then the whole policy on this library should be reviewed: what is callable from foreign threads and what is not...